src/google-calendar-api.js

```diff
@@ -3,28 +3,38 @@ var google = require('googleapis');
 var calendar = google.calendar('v3').events;
 var moment = require('moment');
 
-var credentials = new Promise((resolve, reject) => {
-	const authFactory = new GoogleAuth();
-	const jwtClient = new authFactory.JWT(
-		process.env.GOOGLE_CLIENT_EMAIL,
-		null,
-		process.env.GOOGLE_PRIVATE_KEY,
-		['https://www.googleapis.com/auth/calendar']
-	);
+var credentials = (function() {
+	var pending;
 
-	jwtClient.authorize(error => error ? reject(error) : resolve(jwtClient));
-});
+	return function() {
+		if (!pending) {
+			pending = new Promise((resolve, reject) => {
+				const authFactory = new GoogleAuth();
+				const jwtClient = new authFactory.JWT(
+					process.env.GOOGLE_CLIENT_EMAIL,
+					null,
+					process.env.GOOGLE_PRIVATE_KEY,
+					['https://www.googleapis.com/auth/calendar']
+				);
+
+				jwtClient.authorize(error => error ? reject(error) : resolve(jwtClient));
+			});
+		}
+
+		return pending;
+	};
+})();
 
 module.exports = {
-	checkIfWfhEventExists: employeeName => {
+	checkIfWfhEventExists: (employeeName, eventStartDateTime, eventEndDateTime = new Date(moment(eventStartDateTime).add(1, 'day').startOf('day').format())) => {
```

Comment:
Well, I just learned something: I didn't know you could reference other parameters in default parameters. That's really cool! In transpiled code, this works implicitly, but I was surprised here so much so that I tried it in the console.

Though it's "compact", I would reconsider the readability of this function signature, a lot is happening in a single line. Maybe moving some of this defaulting into the function body itself would improve readability.

Alternatively, you can define the defaulting in a separate function so your signature looks like this:
```
const plusOneDay = date => new Date(moment(date).add(1, 'day').startOf('day').format());
checkIfWfhEventExists(employeeName, eventStartDateTime, eventEndDateTime = plusOneDay(eventStartDateTime)) { /* ... */ }
```

Though even that looks a bit verbose. Ultimately, it's a matter of opinion as these are all equivalent, but readability is important to other developers who aren't as familiar with the code.